// landing-blocks/dist/Button.js
"use strict";
var __makeTemplateObject = (this && this.__makeTemplateObject) || function (cooked, raw) {
    if (Object.defineProperty) { Object.defineProperty(cooked, "raw", { value: raw }); } else { cooked.raw = raw; }
    return cooked;
};
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
/** @jsx jsx */
var react_1 = require("@chakra-ui/react");
var react_2 = require("@emotion/react");
var color_js_1 = __importDefault(require("color-js"));
var react_3 = require("react");
var bouncein_1 = __importDefault(require("./animations/bouncein"));
var headshake_1 = __importDefault(require("./animations/headshake"));
var shake_1 = __importDefault(require("./animations/shake"));
var wobble_1 = __importDefault(require("./animations/wobble"));
var support_1 = require("./support");
var myAnimations = {
    wobble: wobble_1.default,
    bounceIn: bouncein_1.default,
    headShake: headshake_1.default,
    shake: shake_1.default,
};
exports.Button = react_3.forwardRef(function (_a, ref) {
    var _b = _a.bg, bg = _b === void 0 ? 'primary' : _b, animate = _a.animate, props = __rest(_a, ["bg", "animate"]);
    var realBg = support_1.useColor(bg);
    var lightness = color_js_1.default(realBg).getLightness();
    // console.log({ lightness })
    var isDark = lightness < 0.7;
    var color = isDark ? 'white' : 'black';
    // console.log('bg', bg)
    // console.log('color', color)
    if (!props.children) {
        return react_2.jsx(react_3.Fragment, null);
    }
    var animationCss = react_3.useMemo(function () {
        if (animate === true) {
            return makeAnimationCss(bouncein_1.default);
        }
        if (animate && animate in myAnimations) {
            return makeAnimationCss(myAnimations[animate]);
        }
        return react_2.css(templateObject_1 || (templateObject_1 = __makeTemplateObject([""], [""])));
    }, [animate]);
    return (react_2.jsx(react_1.Button, __assign({ css: animationCss, ref: ref, as: props.href ? 'a' : 'button', transition: 'all 0.1s ease-in-out', px: '20px', d: 'block', width: 'auto', color: color, bg: bg, fontSize: 'text', fontWeight: 'medium', _hover: { shadow: 'xl', transform: 'scale(1.1)' } }, props)));
});
function makeAnimationCss(keyframes) {
    return react_2.css(templateObject_2 || (templateObject_2 = __makeTemplateObject(["\n        animation-duration: 0.6s;\n        /* animation-delay: 1s; */\n        animation-timing-function: ease-in-out;\n        animation-name: ", ";\n        animation-direction: normal;\n        /* animation-iteration-count: infinite; */\n    "], ["\n        animation-duration: 0.6s;\n        /* animation-delay: 1s; */\n        animation-timing-function: ease-in-out;\n        animation-name: ", ";\n        animation-direction: normal;\n        /* animation-iteration-count: infinite; */\n    "])), keyframes);
}
var templateObject_1, templateObject_2;
