// landing-blocks/dist/hooks.js
"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (Object.hasOwnProperty.call(mod, k)) result[k] = mod[k];
    result["default"] = mod;
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
var react_1 = require("@chakra-ui/react");
var baby_i_am_faded_1 = require("baby-i-am-faded");
// import { Faded, FadedProps } from 'baby-i-am-faded'
var react_2 = __importStar(require("react"));
require("intersection-observer");
function useFaded(_a) {
    var _b = _a.animate, animate = _b === void 0 ? undefined : _b, _c = _a.triggerOnce, triggerOnce = _c === void 0 ? true : _c, rest = __rest(_a, ["animate", "triggerOnce"]);
    var component = react_2.useMemo(function () {
        var defaultProps = {
            cascade: true,
            threshold: 0.1,
            duration: 400,
            triggerOnce: triggerOnce,
            children: [],
        };
        return function (props) {
            return react_2.default.createElement(baby_i_am_faded_1.Faded, __assign({}, defaultProps, props));
        };
    }, [animate, triggerOnce]);
    return { Faded: animate ? component : 'div' };
}
exports.useFaded = useFaded;
function useMyColorMode(_a) {
    var _b = _a.dark, dark = _b === void 0 ? false : _b;
    var colorMode = react_1.useColorMode().colorMode;
    if (dark) {
        return { colorMode: 'dark' };
    }
    return { colorMode: colorMode };
}
exports.useMyColorMode = useMyColorMode;
